src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -443,9 +443,25 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
   -->
   <Target Name="_GetRestoreSettings"
-          DependsOnTargets="_GetRestoreProjectStyle;_GetRestoreSettingsOverrides"
+          DependsOnTargets="_GetRestoreProjectStyle;_GetRestoreSettingsOverrides;_GetRestoreTargetFrameworksAsItems"
           Returns="$(_OutputSources);$(_OutputPackagesPath);$(_OutputFallbackFolders);$(_OutputConfigFilePaths)">
 
+    <!-- Read additional sources and fallback folders for each framework  -->
+    <MSBuild
+      Condition=" '$(RestoreProjectStyle)' == 'PackageReference' "
```

Comment:
It doesn't need to do inner builds for legacy, but I think it should still read the sources and fallback folders. It would be reading the settings in two places if this special cased legacy.